src/utils/canvas.js

```diff
@@ -4,19 +4,104 @@ import Constants from '../core/Constants';
 
 import * as Colors from './colors';
 
-// given the imageData object returns the DOM Image with loaded data
-function imageData2Image(imagedata) {
+// given a single channel UInt8 image data mask with non-zero values indicating the
+// mask, turn it into a 4 channgel RGBA image data URL filled in with the given
+// color for pixels turned on in the mask.
+function mask2DataURL(singleChannelData, nw, nh, color) {
   const canvas = document.createElement('canvas');
   const ctx = canvas.getContext('2d');
 
-  canvas.width = imagedata.width;
-  canvas.height = imagedata.height;
-  ctx.putImageData(imagedata, 0, 0);
+  canvas.width = nw;
+  canvas.height = nh;
+
+  const numChannels = 1;
+
+  setMaskPixelColors(ctx, singleChannelData, nw, nh, color, numChannels);
+  
+  const url = canvas.toDataURL();
+
+  return url;
+}
+
+// given an RGBA image data URL, turn it into an actual DOM Image filled in with the current
+// class color.
+function maskDataURL2Image(maskDataURL, { color = Constants.FILL_COLOR } = {}) {
+  return new Promise((resolve, _reject) => {
+    const img = document.createElement('img');
+
+    img.onload = () => {
+      const canvas = document.createElement('canvas');
+      const nw = img.width,
+        nh = img.height;
+
+      canvas.width = nw;
+      canvas.height = nh;
+
+      const ctx = canvas.getContext('2d');
+
+      ctx.drawImage(img, 0, 0);
+
+      const imgData = ctx.getImageData(0, 0, nw, nh);
+
+      const numChannels = 4; // RGBA
+
+      setMaskPixelColors(ctx, imgData.data, nw, nh, color, numChannels);
+
+      img.src = canvas.toDataURL();
 
-  const image = new Image();
+      resolve(img);
+    };
+    img.src = maskDataURL;
+  });
+}
+
+// Given some RGBA mask pixel array, efficiently sets the colors. Note that we assume that the same value is set
+// throughout the channels of the mask, so that no channel will be set to 0 if there is a valid mask
+// position there (i.e. all channels might be 255 if a mask is present).
+function setMaskPixelColors(ctx, data, nw, nh, color, numChannels) {
```

Comment:
it's better to document it in jsdoc style to have description for every parameter